### PR TITLE
feat(core): add dynamic error messages

### PR DESCRIPTION
**Pull request by Waidmann**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

One can pass a static errorMessage as string to the ThrottlerModuleOptions or it is defaulted to a static string.

## What is the new behavior?

One can pass either a static errorMessage as string OR a function to generate an error message based on the current context to the ThrottlerModuleOptions or it is defaulted to a static string.

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

I personally think this is important in order to track on which routes and for what users rate limiting occurs.

### CODE REVIEW

**Comment by Waidmann** [comment]
@kkoomen @jmcdo29

**Comment by kkoomen** [comment]
I'm not into this. @jmcdo29 can you have a look?

**Comment by Waidmann** [comment]
Is this good?

**Comment by Waidmann** [comment]
@jmcdo29

File: src/throttler.guard.ts [hunk 1]
```diff
@@ -224,7 +224,11 @@ export class ThrottlerGuard implements CanActivate {
     _throttlerLimitDetail: ThrottlerLimitDetail,
   ): Promise<string> {
     if (!Array.isArray(this.options)) {
-      return this.options.errorMessage || this.errorMessage;
+      if (!this.options.errorMessage) return this.errorMessage;
+
+      return typeof this.options.errorMessage === 'function'
+        ? this.options.errorMessage(_context, _throttlerLimitDetail)
```
Comment (micalevisk) [inline src/throttler.guard.ts:230]: minor refactor: remove the `_` prefix from the parameters as they're being used now

Comment (Waidmann) [reply src/throttler.guard.ts]: Fair!

**Comment by Waidmann** [comment]
@micalevisk and @jmcdo29 does this look good now?

**Review by micalevisk — APPROVED** [review]

**Review by jmcdo29 — APPROVED** [review]
LGTM

**Comment by jmcdo29** [comment]
Appreciate the patience. It can be a bit sufficient to get reviews in during travels

**Pull request merged**